iceoryx2/src/node/mod.rs

```diff
@@ -0,0 +1,709 @@
+            .monitor();
+
+        if let Ok(result) = result {
+            return Self::state_from_monitor(&result);
+        }
+
+        let msg = "Unable to acquire node monitor";
+        let origin = format!("Node::get_node_state({:?}, {:?})", config, name);
+        match result.err().unwrap() {
+            MonitoringCreateMonitorError::InsufficientPermissions => {
+                fail!(from origin, with NodeListFailure::InsufficientPermissions,
+                        "{} due to insufficient permissions while acquiring the node state.", msg);
+            }
+            MonitoringCreateMonitorError::Interrupt => {
+                fail!(from origin, with NodeListFailure::Interrupt,
+                        "{} since an interrupt was received while acquiring the node state.", msg);
+            }
+            MonitoringCreateMonitorError::InternalError => {
+                fail!(from origin, with NodeListFailure::InternalError,
+                        "{} since an internal failure occurred while acquiring the node state.", msg);
+            }
+        }
+    }
+
+    fn open_node_storage(
+        config: &Config,
+        node_name: &FileName,
+    ) -> Result<Option<Service::StaticStorage>, NodeReadStorageFailure> {
+        let details_config = node_details_config::<Service>(config, node_name);
+        let result = <Service::StaticStorage as StaticStorage>::Builder::new(
+            &FileName::new(b"node").unwrap(),
+        )
+        .config(&details_config)
+        .has_ownership(false)
+        .open();
+
+        if let Ok(result) = result {
+            return Ok(Some(result));
+        }
+
+        let msg = "Unable to open node config storage";
+        let origin = format!("open_node_storage({:?}, {:?})", config, node_name);
+
+        match result.err().unwrap() {
+            StaticStorageOpenError::DoesNotExist => Ok(None),
+            StaticStorageOpenError::Read => {
+                fail!(from origin, with NodeReadStorageFailure::ReadError,
+                    "{} since the node config storage could not be read.", msg);
+            }
+            StaticStorageOpenError::IsLocked => {
+                fail!(from origin, with NodeReadStorageFailure::Corrupted,
+                    "{} since the node config storage seems to be uninitialized but the state should always be present.", msg);
+            }
+            StaticStorageOpenError::InternalError => {
+                fail!(from origin, with NodeReadStorageFailure::InternalError,
+                    "{} due to an internal failure while opening the node config storage.", msg);
+            }
+        }
+    }
+
+    fn get_node_details(
+        config: &Config,
+        node_name: &FileName,
+    ) -> Result<Option<NodeDetails>, NodeReadStorageFailure> {
+        let node_storage = Self::open_node_storage(config, node_name)?;
+        if node_storage.is_none() {
+            return Ok(None);
+        }
+        let node_storage = node_storage.unwrap();
```

Comment:
A little bit more idiomatic
```suggestion
        let node_storage = if let Some(n) = Self::open_node_storage(config, node_name)? {
            n
        } else {
            return Ok(None);
        };
```